test/multivariate/mvlogitnormal.jl

```diff
@@ -0,0 +1,163 @@
+# Tests on Multivariate Logit-Normal distributions
+using Distributions
+using ForwardDiff
+using LinearAlgebra
+using Random
+using Test
+
+####### Core testing procedure
+
+function test_mvlogitnormal(d::MvLogitNormal; nsamples::Int=10^6)
+    @test d.normal isa AbstractMvNormal
+    dnorm = d.normal
+
+    @testset "properties" begin
+        @test length(d) == length(dnorm) + 1
+        @test params(d) == params(dnorm)
+        @test partype(d) == partype(dnorm)
+        @test eltype(d) == eltype(dnorm)
+        @test location(d) == mean(dnorm)
+        @test minimum(d) == fill(0, length(d))
+        @test maximum(d) == fill(1, length(d))
+        @test insupport(d, normalize(rand(length(d)), 1))
+        @test !insupport(d, normalize(rand(length(d) + 1), 1))
+        @test !insupport(d, rand(length(d)))
+        x = rand(length(d) - 1)
+        x = vcat(x, -sum(x))
+        @test !insupport(d, x)
+    end
+
+    @testset "conversions" begin
+        @test convert(typeof(d), d) === d
+        T = partype(d) <: Float64 ? Float32 : Float64
+        if dnorm isa MvNormal
+            @test convert(MvLogitNormal{MvNormal{T}}, d).normal ==
+                convert(MvNormal{T}, dnorm)
+            @test partype(convert(MvLogitNormal{MvNormal{T}}, d)) <: T
+            @test convert(MvLogitNormal{MvNormal{T}}, params(d)...).normal ==
+                convert(MvNormal{T}, dnorm)
+            @test partype(convert(MvLogitNormal{MvNormal{T}}, params(d)...)) <: T
+            @test canonform(d) isa MvLogitNormal{<:MvNormalCanon}
+            @test canonform(d).normal == canonform(dnorm)
+        elseif dnorm isa MvNormalCanon
+            @test convert(MvLogitNormal{MvNormalCanon{T}}, d).normal ==
+                convert(MvNormalCanon{T}, dnorm)
+            @test partype(convert(MvLogitNormal{MvNormalCanon{T}}, d)) <: T
+            @test convert(MvLogitNormal{MvNormalCanon{T}}, params(d)...).normal ==
+                convert(MvNormalCanon{T}, dnorm)
+            @test partype(convert(MvLogitNormal{MvNormalCanon{T}}, params(d)...)) <: T
+            @test meanform(d) isa MvLogitNormal{<:MvNormal}
+            @test meanform(d).normal == meanform(dnorm)
+        end
+    end
+
+    @testset "sampling" begin
+        X = rand(d, nsamples)
+        Y = @views log.(X[1:(end - 1), :]) .- log.(X[end, :]')
+        Ymean = vec(mean(Y; dims=2))
+        Ycov = cov(Y; dims=2)
+        for i in 1:(length(d) - 1)
+            @test isapprox(
```

Comment:
I'm not completely sure anymore but I assume we don't have any statistical test utilities for e.g. hypothesis tests of samples? IIRC KS tests are used for the univariate distributions but I assume we don't have anything for multi- or matrixvariate distributions.

Reply:
This is the closest I can find, which just checks that a 1x1 matrix distribution is consistent with the univariate distribution it reduces to:
https://github.com/JuliaStats/Distributions.jl/blob/b21e515dd4684d2d0f2c706c573ed6eb98ce537f/test/matrixvariates.jl#L193-L199

Reply:
Thank you for checking! Would be useful to have something similar for multivariate distributions (generally, IMO we should provide test utilities such that you can check more easily that you implement the interface correctly) but better to do this in a separate PR.